app/controllers/CalculationController.scala

```diff
@@ -83,7 +97,53 @@ class CalculationController @Inject()(viewLPP: CalculationLPPView,
             val additionalPenaltyRate = "4"
             val daysSince31 = ChronoUnit.DAYS.between(penalty.get.period.dueDate.plusDays(31), LocalDateTime.now())
             val isEstimate = penalty.get.status.equals(PointStatusEnum.Estimated)
-            Ok(viewAdd(daysSince31, isEstimate, additionalPenaltyRate, startDateOfPeriod, endDateOfPeriod, penaltyAmount, amountReceived,amountLeftToPay))
+            Ok(viewAdd(daysSince31, isEstimate, additionalPenaltyRate, startDateOfPeriod, endDateOfPeriod, penaltyAmount, amountReceived, amountLeftToPay))
+          }
+        }
+      }
+    }
+  }
+
+  def getPenaltyDetailsFromNewAPI(penaltyId: String, isAdditional: Boolean)(implicit request: User[_]): Future[Result] = {
+    penaltiesService.getPenaltyDetailsFromEnrolmentKey(EnrolmentKeys.constructMTDVATEnrolmentKey(request.vrn)).map {
+      payload => {
+        val penalty: Option[LPPDetails] = payload.latePaymentPenalty.flatMap(_.details.find(_.principalChargeReference == penaltyId))
```

Comment:
This is going to be refactored later to use principalChargeReference & isAdditional flag